php/libraries/DataFrameworkMenu.class.inc

```diff
@@ -183,5 +183,91 @@ abstract class DataFrameworkMenu extends NDB_Menu_Filter
         return new \LORIS\Http\DataIteratorBinaryStream($rows);
 
     }
+
+    /**
+     * Format the list of user's projects to display the module filter options
+     *
+     * @param bool $numericKeys If the returned projects keys
+     *                          should be numeric (ProjectID) or project name
+     *
+     * @return array - an associative array of the form (ID|mame => name)
+     */
+    public function getProjectOptions(
+        bool  $numericKeys = true
+    ): array {
+        $projects     = \NDB_Factory::singleton()->user()->getProjects();
```

Comment:
If you need a user, I think it should be passed as a parameter.

Reply:
I will make this change but I'm curious to know more about the reasoning behind this. It will help if I understand the motivations of using a parameters (and why user should be defined in each module getFieldsOptions instead).

Reply:
Because it's a functional dependency of the method, so it should be explicit. A singleton is a hidden global variable. I think once you get up to getFieldOptions you don't have a choice but to use the singleton since you can't change the signature, but the dependencies should be as explicit as possible (which also makes it easier to inject values for testing.)

Reply:
Thanks for the clarifications.